import {supabase} from '../supabase/supabase.config.jsx'
import {create} from 'zustand'

export const useStorePrestamos = create((set) => ({
    error: null,
    loading: false,
    prestamos: [],
    cuotas: [],

    crearPrestamo: async (prestamo, cuotas = []) => {
        set({ loading: true, error: null });
        try {
        // 1. Insertar pr√©stamo en la tabla PRESTAMO
            const { data: prestamoInsertado, error: errorPrestamo } = await supabase
                .from("PRESTAMOS")
                .insert(prestamo)
                .select()
                .single();

            console.log("Pr√©stamo insertado:", prestamoInsertado);

        if (errorPrestamo) throw errorPrestamo;

        // 2. Insertar cuotas ligadas al pr√©stamo
        if (cuotas.length > 0) {
            const cuotasConFk = cuotas.map((c) => ({
            ...c,
            id_prestamo: prestamoInsertado.id_prestamo, // üëà FK
            }));

            const { error: errorCuotas } = await supabase
            .from("CUOTAS")
            .insert(cuotasConFk);

            if (errorCuotas) throw errorCuotas;
        }

        set((state) => ({
            loading: false,
            error: null,
            prestamos: [prestamoInsertado, ...state.prestamos],
        }));

        console.log("Pr√©stamo y cuotas creados exitosamente", prestamoInsertado, cuotas);

        return prestamoInsertado;
        } catch (err) {
        console.error("Error al crear pr√©stamo:", err);
        set({ loading: false, error: err.message });
        throw new Error(err.message);
        }
    },

    obtenerPrestamosPorUsuario: async (id_cuenta) => {
        set({ loading: true, error: null });
        try {
            // 1. Traer pr√©stamos y sus cuotas relacionadas
            const { data: prestamos, error } = await supabase
            .from("PRESTAMOS")
            .select(`
                *,
                CUOTAS(*)
            `)
            .eq("id_cuenta", id_cuenta); // filtrar por la cuenta del usuario

            if (error) throw error;

            set({ loading: false, prestamos, error: null });
            console.log("Pr√©stamos obtenidos:", prestamos);
            return prestamos;
        } catch (err) {
            console.error("Error al obtener pr√©stamos:", err);
            set({ loading: false, error: err.message });
            throw new Error(err.message);
        }
    },


    }));
